src/servers/openai-mcp-server.ts

```diff
@@ -99,10 +99,21 @@ export class OpenAIDeepResearchMCPServer extends BaseMCPServer {
         // First check if the query is of the form "datasource:<id> <query-with-spaces>. The id is a string of numbers, letters, and hyphens."
         const re = /^(?:datasource:(?<id>[A-Za-z0-9-]+)\s+)?(.+)$/;
         const match = re.exec(query);
-        const datasourceId = match?.groups?.id;
         const queryWithoutDatasourceId = match![2];
-        if (datasourceId) {
-            const relevantQuestions = await this.getThoughtSpotService().getRelevantQuestions(queryWithoutDatasourceId, [datasourceId], "");
+        let datasourceIdToUse = match?.groups?.id;
+        // If the datasource id is not provided, get the data source suggestions given the version of the ThoughtSpot instance is greater than 10.13
+        if (!datasourceIdToUse && this.isVersionGreaterThan1013()) {
+            const dataSources = await this.getThoughtSpotService().getDataSourceSuggestions(query);
+            if (!dataSources || dataSources.length === 0) {
+                return this.createErrorResponse("No data source suggestions found", "No data source suggestions found");
+            }
+            // Use the first (highest confidence) data source for getting relevant questions
+            const primaryDataSource = dataSources[0];
+            datasourceIdToUse = primaryDataSource.header.guid;
+        }
+        // If the datasource id is present, get the relevant questions for the query
+        if (datasourceIdToUse) {
+            const relevantQuestions = await this.getThoughtSpotService().getRelevantQuestions(queryWithoutDatasourceId, [datasourceIdToUse], "");
```

Comment:
We want to let the agent choose the datasource for us. We should return the list of sources as search results. And then the agent will do a search again with the relevant datasource added to the query.

Reply:
done